src/Bicep.Core/TypeSystem/Az/AzResourceTypeProvider.cs

```diff
@@ -125,12 +126,32 @@ private static ObjectType SetBicepResourceProperties(ObjectType objectType, Reso
                 properties = properties.SetItem("resourceGroup", new TypeProperty("resourceGroup", LanguageConstants.String, TypePropertyFlags.DeployTimeConstant));
                 properties = properties.SetItem("subscriptionId", new TypeProperty("subscriptionId", LanguageConstants.String, TypePropertyFlags.DeployTimeConstant));
             }
+
+            var functions = GetTypeBicepFunctions(objectType.Name);
+
             return new NamedObjectType(
                 objectType.Name,
                 objectType.ValidationFlags,
                 isExistingResource ? ConvertToReadOnly(properties.Values) : properties.Values,
                 objectType.AdditionalPropertiesType,
-                isExistingResource ? ConvertToReadOnly(objectType.AdditionalPropertiesFlags) : objectType.AdditionalPropertiesFlags);
+                isExistingResource ? ConvertToReadOnly(objectType.AdditionalPropertiesFlags) : objectType.AdditionalPropertiesFlags, 
+                functions);
+        }
+
+        private static IEnumerable<FunctionOverload> GetTypeBicepFunctions(string name)
```

Comment:
This is the function that we can use to define more functions that can be called on particular types.